servers/src/main/java/tachyon/master/file/meta/InodeTree.java

```diff
@@ -283,8 +283,11 @@ public CreatePathResult createPath(TachyonURI path, CreatePathOptions options)
         traversalResult.getNonPersisted().add(lastInode);
         toPersistDirectories.add(lastInode);
       } else {
-        LOG.info(ExceptionMessage.FILE_ALREADY_EXISTS.getMessage(path));
-        throw new FileAlreadyExistsException(ExceptionMessage.FILE_ALREADY_EXISTS.getMessage(path));
+        if (!(lastInode.isDirectory() && options.isAllowExists())) {
```

Comment:
Shouldn't the condition be `lastInode.isDirectory() && !options.isAllowExists()`?